utils/ChromiumDownloader.js

```diff
@@ -226,19 +223,22 @@ function extractZip(zipPath, folderPath) {
   return new Promise(fulfill => extract(zipPath, {dir: folderPath}, fulfill));
 }
 
-function requestOptions(url, method = 'GET') {
+function httpRequest(url, method, response) {
   /** @type {Object} */
-  const result = URL.parse(url);
-  result.method = method;
+  const options = URL.parse(url);
+  options.method = method;
 
   const proxyURL = getProxyForUrl(url);
   if (proxyURL) {
     /** @type {Object} */
     const parsedProxyURL = URL.parse(proxyURL);
     parsedProxyURL.secureProxy = parsedProxyURL.protocol === 'https:';
 
-    result.agent = new ProxyAgent(parsedProxyURL);
+    options.agent = new ProxyAgent(parsedProxyURL);
   }
 
-  return result;
+  const driver = options.protocol === 'https:' ? require('https') : require('http');
+  const request = driver.request(options, response);
```

Comment:
@JoelEinbinder tsc is [unhappy](https://travis-ci.org/GoogleChrome/puppeteer/jobs/311095356) here: 

> utils/ChromiumDownloader.js(241,19): error TS2349: Cannot invoke an expression whose type lacks a call signature. Type '((options: string | URL | RequestOptions, callback?: (res: IncomingMessage) => void) => ClientReq...' has no compatible call signatures.

How to please it?

Reply:
@aslushnikov the one with the least changes would be:

```
  const driver = options.protocol === 'https:' ? 'https' : 'http';
  const request = require(driver).request(options, response);
```

TypeScript like this!